### PR TITLE
External secret dummy resolver

### PR DESCRIPTION
**Pull request by offerm**

### CODE REVIEW

File: raiden/raiden_event_handler.py [hunk 1]
```diff
@@ -167,6 +168,19 @@ def handle_send_secretrequest(
...
+        )
+        x_secret_hash = sha3(x_secret)
+
+        if secret_request_event.secrethash == x_secret_hash:
+            state_change = ReceiveSecretReveal(
+                x_secret,
+                secret_request_event.recipient,
+            )
+            raiden.handle_and_track_state_change(state_change)
+            return
```
Comment (rakanalh) [inline raiden/raiden_event_handler.py]: This looks like it's test code... please remove

File: raiden/raiden_event_handler.py [hunk 2]
```diff
@@ -210,7 +224,7 @@ def handle_paymentsentsuccess(
         # With the introduction of the lock we should always get
         # here only once per identifier so payment_status should always exist
         # see: https://github.com/raiden-network/raiden/pull/3191
-        payment_status.payment_done.set(True)
+        payment_status.payment_done.set(payment_sent_success_event.secret)
```
Comment (rakanalh) [inline raiden/raiden_event_handler.py]: made comments about this in https://github.com/raiden-network/raiden/pull/3560

**Comment by offerm** [comment]
> I think this PR doesn't have to be merged provided that it only replies with a dummy secret as soon as the `SecretRequest` is received. Are there any other additions that are not in #3560?

Yes, This PR includes the change to the state machine (event handler).

File: raiden/raiden_event_handler.py (hunk 2 above)
Comment (offerm) [reply raiden/raiden_event_handler.py]: Noted

File: raiden/raiden_event_handler.py (hunk 1 above)
Comment (offerm) [reply raiden/raiden_event_handler.py]: Suggest you ignore this PR and jump directly to #3578 .

Thanks and sorry for the confusion

**Comment by offerm** [comment]
Will open a new PR once changes to the resolver API  implemented

**Pull request closed without merging**